### PR TITLE
Deprecate `observables` argument in `QubitDevice.statistics`

### PR DESCRIPTION
**Pull request by AlbertMitjans**
Add a deprecation warning when using `QubitDevice.statistics` with a list of observables, or when using the `observables` argument:
```pycon
>>> dev = qml.device("default.qubit", wires=0)
>>> dev.statistics([])
/Users/albertmitjans/Developer/pennylane/pennylane/_qubit_device.py:724: UserWarning: The ``observables`` argument in ``QubitDevice.statistics`` is deprecated. Please use ``circuit`` instead.
  warnings.warn(
>>> dev.statistics(observables=[])
/Users/albertmitjans/Developer/pennylane/pennylane/_qubit_device.py:724: UserWarning: The ``observables`` argument in ``QubitDevice.statistics`` is deprecated. Please use ``circuit`` instead.
  warnings.warn(
```

### CODE REVIEW

File: pennylane/_qubit_device.py [hunk 1]
```diff
@@ -669,20 +669,21 @@ def active_wires(operators):
...
+        self, circuit: QuantumScript = None, shot_range=None, bin_size=None, observables=None
+    ):
         """Process measurement results from circuit execution and return statistics.
 
         This includes returning expectation values, variance, samples, probabilities, states, and
         density matrices.
 
         Args:
-            observables (List[.Observable]): the observables to be measured
+            circuit (~.tape.QuantumScript): the quantum tape currently being executed
```
Comment (antalszava) [inline pennylane/_qubit_device.py:681]:
```suggestion
            circuit (~.tape.QuantumScript): the quantum script currently being executed
```

File: pennylane/_qubit_device.py [hunk 2]
```diff
@@ -716,7 +717,17 @@ def statistics(self, circuit: QuantumScript, shot_range=None, bin_size=None):
             * Finally, we repeat the measurement statistics for the final 100 shots,
               ``shot_range=[35, 135]``, ``bin_size=100``.
         """
-        observables = circuit.observables
+        if isinstance(circuit, QuantumScript):
```
Comment (antalszava) [inline pennylane/_qubit_device.py:720]: Could this be a tape too?

File: pennylane/_qubit_device.py [hunk 3]
```diff
@@ -841,7 +852,7 @@ def _statistics_new(self, circuit: QuantumScript, shot_range=None, bin_size=None
         density matrices.
 
         Args:
-            observables (List[.Observable]): the observables to be measured
+            circuit (~.tape.QuantumScript): the quantum tape currently being executed
```
Comment (antalszava) [inline pennylane/_qubit_device.py:855]:
```suggestion
            circuit (~.tape.QuantumScript): the quantum script currently being executed
```

**Review by antalszava — APPROVED** [review]
Looks good! :tada: Thanks for adding this.

**Comment by AlbertMitjans** [comment]
@trbromley @josh146 waiting for your approval before merging

**Pull request merged**